tests/MaxMind/Test/MinFraudTest.php

```diff
@@ -483,6 +482,105 @@ public function badIps()
     }
 
     /**
+     * @expectedException MaxMind\Exception\InvalidInputException
+     * @expectedExceptionMessage must be greater than or equal to 0
+     * @dataProvider negativeSessionAge
+     */
+    public function testNegativeSessionAge($age)
+    {
+        $this->createMinFraudRequestWithFullResponse(
+            'insights',
+            0
+        )->withDevice(['ip_address' => '1.2.3.4', 'session_age' => $age]);
+    }
+
+    public function negativeSessionAge()
+    {
+        return [
+            [-1],
+        ];
+    }
+
+    /**
+     * @expectedException MaxMind\Exception\InvalidInputException
+     * @expectedExceptionMessage must be a float number
+     * @dataProvider badSessionAge
+     */
+    public function testBadSessionAge($age)
+    {
+        $this->createMinFraudRequestWithFullResponse(
+            'insights',
+            0
+        )->withDevice(['ip_address' => '1.2.3.4', 'session_age' => $age]);
+    }
+
+    public function badSessionAge()
+    {
+        return [
+            ["X"],
```

Comment:
I'm doing that in `testNegativeSessionAge`.  Is that an appropriate place to do it?